compiler/il/OMRIL.cpp

```diff
@@ -812,6 +639,11 @@ OMR::IL::opCodeForIndirectStore(TR::DataType dt)
 TR::ILOpCodes
 OMR::IL::opCodeForIndirectWriteBarrier(TR::DataType dt)
    {
+   static_assert(TR::NumOMRTypes == (sizeof(OMR::IL::opCodesForIndirectWriteBarrier) / sizeof(OMR::IL::opCodesForIndirectWriteBarrier[0])),
+              "OMR::IL::opCodesForIndirectWriteBarrier is not the correct size");
+
+   if (dt.isVector()) return TR::BadILOp;
```

Comment:
Similar question as for the direct write barrier

Reply:
The same.